include/aikido/perception/DetectedObject.hpp

```diff
@@ -33,34 +34,46 @@ namespace perception {
 class DetectedObject
 {
 public:
+
+  /// Default constructor.
+  DetectedObject() = default;
+
   /// Construct a \c DetectedObject
   /// \param[in] uid Unique ID for object in Aikido world. Same UID -> Same
   /// Object
-  /// \param[in] assetKey Key for AssetDatabase passed into constructor of
```

Comment:
Add documentation for `objectName`, say that it is default Asset Key

Reply:
`objectName` is actually _not_ the default AssetKey. I updated the class docstring to explain this. The way how the latest marker API work is that its namespace is the object name, which may not be the asset key. For example, object name may be "cantaloupe" while db_key is "food_item". (This is useful because we want the uids to be `cantaloupe_1, cantaloupe_2, ...,` instead of 'fooditem_1, ..., `, in the demo code, so that we can filter by the object name. For this reason, we cannot use objectName as the default asset key (it would not exist in the `pr_assets` db). I also changed the constructor implementation to throw and error if `db_key` is not provided.